### PR TITLE
RUN-2321-added unit test for date time picker

### PR DESCRIPTION
**Pull request by jayas006**
**Is this a bugfix, or an enhancement? Please describe.**
This is an enhancement. We've improved the unit tests for the DateTimePicker component to ensure more robust and comprehensive coverage

**Describe the solution you've implemented**
 I have added and updated unit tests for the DateTimePicker component. These tests now cover v-model binding, class binding, attribute setting for both date-picker and time-picker, CSS class and prop existence, correct datetime computation, handling of edge cases like leap years, and accessibility roles

**Describe alternatives you've considered**

**Additional context**

### CODE REVIEW

File: rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/dateTimePicker.spec.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  it("sets initial data correctly", async () => {
+    const wrapper = await mountDateTimePicker();
+    expect(wrapper.vm.dateString).toBe(now.format("YYYY-MM-DD"));
+    expect(wrapper.vm.time).toBeInstanceOf(Date);
+  });
+  it("emits update:modelValue when time changes", async () => {
+    const wrapper = await mountDateTimePicker();
+    const newTime = new Date();
+    const newTimeFormatted = moment(newTime).format();
+    await wrapper.setData({ time: newTime });
```
Comment (smartinellibenedetti) [inline rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/dateTimePicker.spec.ts:37]: recommend replacing the wrapper.setData with an emit from timePicker (`update:modelValue`), ie:

timePicker.vm.$emit(`update:modelValue`, newTime)

File: rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/dateTimePicker.spec.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    const emitted = wrapper.emitted("update:modelValue");
+    expect(emitted).toBeTruthy();
+    if (emitted) {
+      expect(emitted[0]).toEqual([newTimeFormatted]);
+    }
```
Comment (smartinellibenedetti) [inline rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/dateTimePicker.spec.ts:42]: please remove the if, but keep the line 41.

**Review by smartinellibenedetti — CHANGES_REQUESTED** [review]
Left some pointers, but recommend revisiting these tests, as a few of them seem to be redundant.

File: rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/dateTimePicker.spec.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    await wrapper.setData({ dateString: newDateString });
+    expect(moment(wrapper.vm.time).format("YYYY-MM-DD")).toBe(newDateString);
+  });
+  it("binds v-model, class correctly and sets correct attributes on date-picker and time-picker", async () => {
+    const wrapper = await mountDateTimePicker();
+    const datePicker = wrapper.findComponent({ name: "date-picker" });
+    const timePicker = wrapper.findComponent({ name: "time-picker" });
+    // Check v-model and class for date-picker
+    expect(datePicker.attributes("modelvalue")).toBe(wrapper.vm.dateString);
+    expect(datePicker.attributes("class")).toContain(wrapper.vm.dateClass);
```
Comment (smartinellibenedetti) [inline rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/dateTimePicker.spec.ts:54]: suggest to change `wrapper.vm.dateClass` (both here and on the timepicker for the written class. It's more readable.

File: rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/dateTimePicker.spec.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    const receivedTime = new Date(
+      timePicker.attributes("modelvalue")
+    ).toISOString();
+    const expectedTime = new Date(wrapper.vm.time).toISOString();
+    expect(receivedTime).toBe(expectedTime);
+    expect(timePicker.attributes("class")).toContain(wrapper.vm.timeClass);
+    // Check CSS classes and props
+    expect(wrapper.find(".date-class").exists()).toBe(true);
+    expect(wrapper.find(".time-class").exists()).toBe(true);
+    expect(wrapper.find(".bs-date-picker").exists()).toBe(true);
```
Comment (smartinellibenedetti) [inline rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/dateTimePicker.spec.ts:66]: Can remove those, they are redundant

**Review by smartinellibenedetti — APPROVED** [review]
Left 2 small tweaks but once you commit them, feel free to merge this PR.

**Comment by jayas006** [comment]
> Left 2 small tweaks but once you commit them, feel free to merge this PR.

Fixed

**Comment by smartinellibenedetti** [comment]
there are some typescript errors that need to be adjusted in this unit tests
